starting over

maiamcc opened the pull request:
Porting most of @cjc8's logic, but for me to make it pythonic/testable it's easier if i do it in a fresh file instead of changing it in place.

Here, I've got a new auth flow using env variables instead of a credentials file; the tweet listener and high-level handler flow according to #2; and some tests.

mNov commented on README.md:
```diff
@@ -1 +1,33 @@
+
+## Setup
+- Install requirements: `pip install -r requirements.txt`
+- Make sure you have your auth info for your bot account exported as environment variables
```
maybe would be helpful to link to the Twitter bot documentation? and start with that step, "create a Twitter bot account" or something - just to make it more explicit.

mNov commented on README.md:
```diff
@@ -1 +1,33 @@
+    3. start a new terminal window or in the current one, run `source ~/.bash_profile` (or `~/.bashrc` or wherever you made the change) 
+### About your bot account setup
+- your bot needs "elevated" access
+- your bot need read/write access: user authentication settings > OAuth 1.0a > app permissions, select "read and write". (The required URL values below can just be nonsense.)
```
nit - typo, should be "needs"

mNov commented on README.md:
```diff
@@ -1 +1,33 @@
+
+If given a name, will search the DB (using substring search, i.e. searching for `WHERE name LIKE '%name%'`). If the name uniquely matches a row, return that row; otherwise print info about all matching names with their badge numbers, and request user to search by unique name or badge.
+
+Any request that isn't obviously a badge number (i.e. isn't purely numeric) will be treated as a name query.
```
very helpful readme!

mNov commented on bot.py:
```diff
@@ -0,0 +1,112 @@
+import tweepy
+import urllib
+
+API_URL = os.environ.get('COP_API_URL', 'http://localhost:3000/api/v1/cops')
```
should there be something about this in the README too (i.e. running the API server)? maybe not, i'm not sure if that's a good idea/necessary

maiamcc replied:
that's... a great question. I will put in a line about running it, but I'm hesitant to explicitly link to the other repo; @schrockblock do you have thoughts about this?

maiamcc replied:
decided in meeting that this is okay, will put it in a future PR

mNov commented on bot.py:
```diff
@@ -0,0 +1,112 @@
+    def on_status(self, status: Status):
+        """Extends tweepy.Stream, will be called on every incoming Tweet"""
+        if status.author.id != self.user.id:  # only respond to tweets from users that are not the bot
+            print(f'Got tweet from @{status.author.screen_name}: {status.text}')
```
where will this be printed?

maiamcc replied:
on whatever server we're running the bot. We can decide before we actually host this somewhere that they're more annoying than helpful, but I want to leave them for now for debugging purposes. In a future PR I can put them in a `logger` object that supports multiple log levels, that'll take like 2 minutes.

maiamcc replied:
(But like, whoever is tweeting at us won't see this, if that's your question

mNov replied:
that makes sense, I was wondering if this was for logging/debugging. sounds good!

mNov commented on test_bot.py:
```diff
@@ -0,0 +1,86 @@
+    'TWITTER_ACCESS_TOKEN_SECRET': 'quux',
+}
+
+
```
great tests here!

maiamcc commented:
@mNov think I addressed all your feedback, lmk if there's anything else. If not, do you feel good about approving? (Or did you want to wait for someone else to review?)

mNov commented on README.md:
```diff
@@ -1 +1,36 @@
+
+## Setup
+- Install requirements: `pip install -r requirements.txt`
+- [Register for Twitter developer access](https://developer.twitter.com/en/portal/petition/essential/basic-info) and create a new bot with an api key and secret, and an access token and secret (the first two should be generated for you on bot creation; you may need to click a button to generate the latter two)
```
nice 👍

mNov commented on README.md:
```diff
@@ -1 +1,36 @@
+
+### About testing your bot
+- whatever account is tweeting at your bot may NOT be private/have its tweets protected, or else the API won't be able to pick up its tweets to respond (even if that account and your bot account follow each other)
+- when you're done testing, you can run `./clean_up.py` to delete all tweets from your bot.
```
this is great, btw!

cjc8 replied:
Really really great!!

mNov commented on clean_up.py:
```diff
@@ -0,0 +1,29 @@
+deleted = 0
+failed = 0
+
+for status in tweepy.Cursor(api.user_timeline).items():
```
very cool feature here 👍

cjc8 replied:
This was the worst part of testing the twitter bot so kudos to you for figuring this out

mNov reviewed (approved):
looks good to me! still wondering about whether to link the 2 repos, but that can be addressed later

cjc8 reviewed (approved):
LGTM!
